### DISCUSSION TITLE
Opening preferences (⌘ + ,) causes a quick window flash in macOS

### DISCUSSION THREAD

**Discussion by skarekrow** [kovidgoyal/kitty#5358] [Q&A] [answered]
Hello!

Loving the terminal, but I'm seeing an issue where Kitty will not open its preferences when launched as a GUI application. If I run kitty within kitty, this behavior works as expected. My VISUAL and EDITOR environment variables are set.

Thanks for any help :)

**Comment by page-down** [comment, marked as answer]
When you press `cmd+,`, the editor will be run directly by kitty, not your shell, so the kitty's own environment variables are used.
Try configuring `editor` as an absolute path, or configure `exe_search_path`.

https://sw.kovidgoyal.net/kitty/faq/#things-behave-differently-when-running-kitty-from-system-launcher-vs-from-another-terminal

https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.editor
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.exe_search_path

If you use environment variables in your editor configuration file, you can configure them with `env` if the editor won't start due to missing ones, because what you configure in the shell won't be used either.
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.env

**Reply by skarekrow** [reply]
Thanks! I tried setting the `editor` setting in kitty but that made no difference, even as an absolute path. But setting `PATH` as an env var in the config did the trick, just needed to add `/usr/local/bin`. I wish I was able to see that debug when this issue happened instead of shooting in the dark because the window went away instantly, but I'll gladly take a PEBKAC solution.

Thanks for your help!
